src/Analyzer.JsonRuleEngine/JsonRuleEngine.cs

```diff
@@ -49,6 +49,57 @@ public static JsonRuleEngine Create(string rawRuleDefinitions, BuildILineNumberR
             return new JsonRuleEngine(ParseRuleDefinitions(rawRuleDefinitions), jsonLineNumberResolverBuilder);
         }
 
+        /// <summary>
+        /// Modifies the rules to run based on values defined in the configurations file.
+        /// </summary>
+        /// <param name="configuration">The template configurations to filter and analyze.</param>
+        public void FilterRules(string configuration)
+        {
+            var filteredRules = new List<RuleDefinition>();
+
+            ConfigurationDefinition contents;
+            try
+            {
+                contents = JsonConvert.DeserializeObject<ConfigurationDefinition>(configuration);
+            }
+            catch (Exception e)
+            {
+                throw new JsonRuleEngineException($"Failed to parse configurations file.", e);
+            }
+
+            if (contents.InclusionsConfigurationDefinition != null)
+            {
+                var includeSeverities = contents.InclusionsConfigurationDefinition.Severity;
+                var includeIds = contents.InclusionsConfigurationDefinition.Ids;
+                foreach (RuleDefinition rule in RuleDefinitions)
+                {
+                    if (includeSeverities.Contains(rule.Severity) || includeIds.Contains(rule.Id))
+                    {
+                        filteredRules.Add(rule);
+                    }
+                }
```

Comment:
Suggestion (I think this will work as-is, might need slight modification).  Could also do something similar in `else` block.
``` suggestion
                RuleDefinitions = RuleDefinitions.Where(r => includeSeverities.Contains(r.Severity) || includeIds.Contains(r.Id)).ToList().AsReadOnly();
```